### PR TITLE
mvwm-git: fix cross-build

### PR DESCRIPTION
**Pull request by ndowens**

### CODE REVIEW

File: srcpkgs/mvwm-git/patches/0001-Fix-cross-build.patch [hunk 1]
```diff
@@ -0,0 +1,177 @@
+From 53b028752d6fdd96471be8cd4c3d1c8d490df2a5 Mon Sep 17 00:00:00 2001
+From: Nathan Owens <ndowens@artixlinux.org>
+Date: Fri, 25 Dec 2020 19:45:56 -0600
+Subject: [PATCH] Fix cross-build
```
Comment (ericonr) [inline srcpkgs/mvwm-git/patches/0001-Fix-cross-build.patch:4]: Have you tried to upstream this? Would be nice to not have to rebase and add new changes when there are updates.

Comment (ndowens) [reply srcpkgs/mvwm-git/patches/0001-Fix-cross-build.patch:4]: Not yet. Add this to my TODO list :)

Comment (ericonr) [reply srcpkgs/mvwm-git/patches/0001-Fix-cross-build.patch:4]: Eh, upstream seems inactive since 2015... I'd consider dropping.

Comment (ericonr) [reply srcpkgs/mvwm-git/patches/0001-Fix-cross-build.patch:4]: Still, the patch likely won't become stale, so it wouldn't be so bad to add it.

File: srcpkgs/mvwm-git/patches/0001-Fix-cross-build.patch [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+-remove all offending options like -Werror from the CFLAGS and
+-CPPFLAGS variables and run configure again."))
+ 
+ # check size of some types
+ ac_save_CFLAGS="$CFLAGS"
+@@ -540,22 +530,8 @@ if test ! x"$png_LIBS" = x; then
+   my_LIBS="$LIBS"
+   CPPFLAGS="$CPPFLAGS $png_CFLAGS"
+   LIBS="$LIBS $png_LIBS -lz -lm"
+-  AC_TRY_RUN([#include <png.h>
```
Comment (sgn) [inline srcpkgs/mvwm-git/patches/0001-Fix-cross-build.patch:35]: I would try to minimize that patch we have. Let's say by just change `AC_TRY_RUN` to `AC_TRY_COMPILE`

Comment (ndowens) [reply srcpkgs/mvwm-git/patches/0001-Fix-cross-build.patch:35]: Doing that causes:
`configure.ac:113: error: possibly undefined macro: AC_DEFINE
      If this token and others are legitimate, please use m4_pattern_allow.`

**Pull request closed without merging**